src/awkward/_v2/contents/listarray.py

```diff
@@ -425,3 +426,91 @@ def _localindex(self, axis, depth):
                 self._identifier,
                 self._parameters,
             )
+
+    def _combinations(self, n, replacement, recordlookup, parameters, axis, depth):
+        if n < 1:
+            raise ValueError("in combinations, 'n' must be at least 1")
+        posaxis = self._axis_wrap_if_negative(axis)
+        if posaxis == depth:
+            return self._combinations_axis0(n, replacement, recordlookup, parameters)
+        elif posaxis == depth + 1:
+            if (
+                self.parameter("__array__") == '"string"'
+                or self.parameter("__array__") == '"bytestring"'
+            ):
+                raise ValueError(
+                    "ak.combinations does not compute combinations of the characters of a string; please split it into lists"
+                )
+
+            totallen = ak._v2.index.Index64.empty(1, self.nplike, dtype=np.int64)
+            offsets = ak._v2.index.Index64.empty(
+                len(self) + 1, self.nplike, dtype=np.int64
+            )
+            self._handle_error(
+                self.nplike[
+                    "awkward_ListArray_combinations_length",
+                    totallen.to(self.nplike).dtype.type,
+                    offsets.to(self.nplike).dtype.type,
+                    self._starts.to(self.nplike).dtype.type,
+                    self._stops.to(self.nplike).dtype.type,
+                ](
+                    totallen.to(self.nplike),
+                    offsets.to(self.nplike),
+                    n,
+                    replacement,
+                    self._starts.to(self.nplike),
+                    self._stops.to(self.nplike),
+                    len(self),
+                )
+            )
+
+            nplike_tocarryraw = self.nplike.empty(n, dtype=np.intp)
+            tocarry = []
+
+            for i in range(n):
+                ptr = ak._v2.index.Index64.empty(
+                    totallen[0], self.nplike, dtype=np.int64
+                )
+                tocarry.append(ptr)
+                nplike_tocarryraw[i] = ptr.ptr
+
+            tocarryraw = ak._v2.contents.numpyarray.NumpyArray(nplike_tocarryraw)
+            toindex = ak._v2.index.Index64.empty(n, self.nplike, dtype=np.int64)
+            fromindex = ak._v2.index.Index64.empty(n, self.nplike, dtype=np.int64)
+
+            self._handle_error(
+                self.nplike[
+                    "awkward_ListArray_combinations",
+                    np.int64,
+                    toindex.to(self.nplike).dtype.type,
+                    fromindex.to(self.nplike).dtype.type,
+                    self._starts.to(self.nplike).dtype.type,
+                    self._stops.to(self.nplike).dtype.type,
+                ](
+                    tocarryraw.data,
```

Comment:
Same point about `nplike_tocarryraw` and `tocarryraw`. I'm just trying to reduce indirection so that this will be easier to read.